src/common/util/src/os/lin/lin_mmap_object.cpp

```diff
@@ -55,29 +55,35 @@ class HandleHolder {
     }
 };
 
-class MapHolder {
+class MapHolder : public MappedMemory {
     void* m_data = MAP_FAILED;
     size_t m_size = 0;
     HandleHolder m_handle;
 
 public:
     MapHolder() = default;
 
-    void set(const std::string& path) {
+    void set(const std::string& path, const size_t data_size = 0, const size_t offset = 0) {
         int prot = PROT_READ;
         int mode = O_RDONLY;
         struct stat sb = {};
         m_handle = HandleHolder(open(path.c_str(), mode));
-        OPENVINO_ASSERT(m_handle.get() != -1,
-                        "Can not open file ",
-                        path,
-                        " for mapping. Ensure that file exists and has appropriate permissions");
-        OPENVINO_ASSERT(fstat(m_handle.get(), &sb) != -1, "Can not get file size for ", path);
-        m_size = sb.st_size;
+        if (m_handle.get() == -1) {
+            throw std::runtime_error("Can not open file " + path +
+                                     " for mapping. Ensure that file exists and has appropriate permissions");
+        }
+        if (fstat(m_handle.get(), &sb) == -1) {
+            throw std::runtime_error("Can not get file size for " + path);
+        }
+        m_size = data_size > 0 ? data_size : sb.st_size - offset;
```

Comment:
Please check that `sb.st_size >= offset`

Reply:
Should not we raise an exception in this case because we expect less data in this case that is insufficient for Tensor creation?